device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -462,6 +471,14 @@ void registerMultiplexingConnectionStateCallback(IotHubConnectionStatusChangeCal
     @Override
     public void execute(IotHubConnectionStatus status, IotHubConnectionStatusChangeReason statusChangeReason, Throwable throwable, Object callbackContext)
     {
+        log.trace("DeviceIO notified of status {} with reason {}", status, statusChangeReason);
+
+        if (status == this.state)
+        {
+            // no change in status, so no need to start/stop worker threads.
+            return;
```

Comment:
Again, while not completely necessary, it helps to keep unusual code paths from causing issues. Typically the status updates only flip from DISCONNECTED/DISCONNECTED_RETRYING to CONNECTED. If a bug in the transport layer causes multiple CONNECTECT status updates to fire, then this layer won't needlessly close and re-open the worker thread pools.